Functions/xASL_quant_Basil.m

```diff
@@ -0,0 +1,231 @@
+function [CBF_nocalib] = xASL_quant_Basil(PWI, x)
+%xASL_quant_Basil Perform quantification using FSL BASIL
+% FORMAT: [CBF_nocalib] = xASL_quant_Basil(PWI, x)
+% 
+% INPUT:
+%   PWI             - image matrix of perfusion-weighted image (REQUIRED)
+%   x               - struct containing pipeline environment parameters (REQUIRED)
+%
+% OUTPUT:
+% CBF_nocalib       - Quantified CBF image
+% -----------------------------------------------------------------------------------------------------------------------------------------------------
+% DESCRIPTION: This script performs quantification of the PWI using the FSL Basil pipeline. Final calibration to
+%              physiological units is performed by dividing the quantified PWI by the M0 image/value
+% -----------------------------------------------------------------------------------------------------------------------------------------------------
+% EXAMPLE: CBF_nocalib = xASL_quant_Basil(PWI, x);
+% __________________________________
+% Copyright 2015-2021 ExploreASL
+
+    
+    
+    
+    fprintf('%s\n','Quantification CBF using FSL Basil:');
+    
+    
+    %% Define paths
+    % NEEDS TO BE MADE A RELATIVE PATH TO BE SPACE-INDEPENDENT
+    pathBasilInput = fullfile(x.SESSIONDIR, 'PWI4D_BasilInput.nii');
+    pathBasilOptions = fullfile(x.SESSIONDIR, 'Basil_ModelOptions.txt');
+    dirBasilOutput = fullfile(x.SESSIONDIR, 'BasilOutput');
+    
+    %% Write the PWI as Nifti file for Basil to read as input
+    % FIXME would be good to have a brain mask too at this point
+    
+    xASL_io_SaveNifti(x.P.Path_PWI4D, pathBasilInput, PWI, [], 0);
+
+
+    %% Create option_file that contains options which are passed to Fabber
+    % basil_options is a character array containing CLI args for the Basil command
+    bMultiPLD = false; % for now
```

Comment:
I see, we go for single-TI only just for testing - that is fine, but should be described upfront.